apps/music/views/albums/view.js

```diff
@@ -27,20 +27,31 @@ var AlbumsView = View.extend(function AlbumsView() {
   });
 
   this.searchResults.getItemImageSrc = (item) => this.getThumbnail(item.name);
+  this.client.on('databaseChange', () => this.update());
+
+  this.configureList();
+  this.update();
+});
+
+AlbumsView.prototype.configureList = function() {
+  var list = this.list;
 
+  // Scroll search out of view, even when
+  // there arent' enough list items to scroll.
   this.list.scrollTop = this.searchBox.HEIGHT;
   this.list.minScrollHeight = `calc(100% + ${this.searchBox.HEIGHT}px)`;
 
   this.list.configure({
-    getItemImageSrc: (item) => {
-      return this.getThumbnail(item.name);
-    }
+    getItemImageSrc: (item) => this.getThumbnail(item.name)
   });
 
-  this.client.on('databaseChange', () => this.update());
-
-  this.update();
-});
+  // Show the view only when list has something rendered,
+  // this prevents Gecko painting unnecessarily.
+  list.addEventListener('rendered', function fn() {
```

Comment:
We should probably add a `once()` function to the base view.js for doing this. Then you could do something like:

```
this.once(this.list, 'rendered', () => document.body.hidden = false);
```